packages/core/src/browser/status-bar/status-bar.ts

```diff
@@ -70,36 +70,35 @@ export class StatusBarImpl extends VirtualWidget implements StatusBar {
 
     protected backgroundColor: string | undefined;
     protected entries: Map<string, StatusBarEntry> = new Map();
-    protected attached = new Deferred<void>();
 
     constructor(
         @inject(CommandService) protected readonly commands: CommandService,
-        @inject(LabelParser) protected readonly entryService: LabelParser
+        @inject(LabelParser) protected readonly entryService: LabelParser,
+        @inject(FrontendApplicationStateService) protected readonly applicationStateService: FrontendApplicationStateService
     ) {
         super();
         delete this.scrollOptions;
         this.id = 'theia-statusBar';
     }
 
-    onAfterAttach(msg: Message): void {
-        super.onAfterAttach(msg);
-        this.attached.resolve();
+    protected get ready(): Promise<void> {
+        return this.applicationStateService.reachedAnyState('initialized_layout', 'ready');
```

Comment:
Why is it necesary to wait for either `'initialized_layout'` or `'ready'`?  Wouldn't it be necessary to just wait for `'initialized_layout'`?

Reply:
This method is also called when the status bar is changed while the application is running. In that case the state is `ready`.